libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/components/PickerBottomSheet.kt

```diff
@@ -0,0 +1,631 @@
+                    unfocusedContainerColor = Color.Transparent,
+                    unfocusedTextColor = colorScheme.secondary,
+                    cursorColor = colorScheme.tertiary,
+                ),
+            )
+            // Url input field
+            OutlinedTextField(
+                value = url,
+                onValueChange = { url = it },
+                label = { Text(stringResource(R.string.sf__server_url_default_custom_url)) },
+                singleLine = true,
+                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Uri),
+                modifier = Modifier.fillMaxWidth()
+                    .padding(start = PADDING_SIZE.dp, end = PADDING_SIZE.dp),
+                colors = TextFieldDefaults.colors(
+                    focusedIndicatorColor = colorScheme.tertiary,
+                    focusedLabelColor = colorScheme.tertiary,
+                    focusedTextColor = colorScheme.onSecondary,
+                    focusedContainerColor = Color.Transparent,
+                    unfocusedIndicatorColor = colorScheme.hintTextColor,
+                    unfocusedLabelColor = colorScheme.hintTextColor,
+                    unfocusedContainerColor = Color.Transparent,
+                    unfocusedTextColor = colorScheme.secondary,
+                    cursorColor = colorScheme.tertiary,
+                ),
+            )
+        }
+
+        val trimmedName = name.trim()
+        val trimmedUrl = url.trim()
+        val serverUrl = getValidServer?.let { it(trimmedUrl) }
+        val validInput = trimmedName.isNotBlank() && serverUrl != null
+        Button(
+            modifier = Modifier
+                .padding(PADDING_SIZE.dp)
+                .fillMaxWidth(),
+            shape = RoundedCornerShape(CORNER_RADIUS.dp),
+            contentPadding = PaddingValues(PADDING_SIZE.dp),
+            colors = ButtonColors(
+                containerColor = colorScheme.tertiary,
+                contentColor = colorScheme.tertiary,
+                disabledContainerColor = colorScheme.surfaceVariant,
+                disabledContentColor = colorScheme.surfaceVariant,
+            ),
+            enabled = validInput,
+            onClick = { addNewLoginServer?.let { it(trimmedName, serverUrl!!) } },
+        ) {
+            Text(
+                text = stringResource(R.string.sf__server_url_save),
+                fontWeight = if (validInput) FontWeight.Normal else FontWeight.Medium,
+                color = if (validInput) colorScheme.onPrimary else colorScheme.onErrorContainer,
+            )
+        }
+    }
+
+    LaunchedEffect(Unit) { focusRequester.requestFocus() }
+}
+
+// Ensure no duplicates in the list because they are not allowed by lazy column.
+fun List<Any>.pickerDistinctBy() : List<Any> {
+    return distinctBy { listItem ->
+        when(listItem) {
+            is LoginServer -> with(listItem) { "$name$url" }
+            else -> listItem.toString()
+        }
+    }
+}
```

Comment:
This is probably necessary/overkill since the lazy column uses the list items's toString as a unique key but I would rather just not show duplicate servers at all if they are already stored on the users device pre-13.0.